### PR TITLE
x-goog-api-client in REST API wrappers

### PR DESCRIPTION
**Pull request by jskeet**
Although this will *build* without an update, we shouldn't merge it without also updating to a version of GAX with https://github.com/googleapis/gax-dotnet/pull/345 merged.

### CODE REVIEW

File: apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/AdvancedTranslationClient.cs [hunk 1]
```diff
@@ -357,17 +357,8 @@ public static AdvancedTranslationClient Create(GoogleCredential credential = nul
...
-            var service = new TranslateService(new BaseClientService.Initializer
-            {
-                HttpClientInitializer = scopedCredentials,
-                ApplicationName = AdvancedTranslationClientImpl.ApplicationName,
-                ApiKey = apiKey
-            });
-
-            return new AdvancedTranslationClientImpl(service, model);
-        }
+        private static AdvancedTranslationClient CreateImpl(GoogleCredential scopedCredentials, string apiKey, string model) =>
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/AdvancedTranslationClient.cs]: This doesn't appear to use the `model` parameter any more.

File: apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationClient.cs [hunk 2]
```diff
@@ -358,17 +358,8 @@ public static TranslationClient Create(GoogleCredential credential = null, Trans
...
-            var service = new TranslateService(new BaseClientService.Initializer
-            {
-                HttpClientInitializer = scopedCredentials,
-                ApplicationName = TranslationClientImpl.ApplicationName,
-                ApiKey = apiKey
-            });
-
-            return new TranslationClientImpl(service, model);
-        }
+        private static TranslationClient CreateImpl(GoogleCredential scopedCredentials, string apiKey, TranslationModel model) =>
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationClient.cs]: This doesn't appear to use the `model` parameter any more.

Comment (jskeet) [reply apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationClient.cs]: Eek! Well caught. Will fix :)

**Comment by jskeet** [comment]
Fixed in the last commit. (This is relatively hard to test due to the credential...)

**Comment by jskeet** [comment]
Merging now - it doesn't actually need the GAX change.

**Pull request merged**